crates/orchestrator/src/config.rs

```diff
@@ -159,25 +159,140 @@ async fn build_da_client() -> Box<dyn DaClient + Send + Sync> {
+    da_client: Option<Box<dyn DaClient>>,
+    /// The service that produces proof and registers it onchain
+    prover_client: Option<Box<dyn ProverClient>>,
+    /// Settlement client
+    settlement_client: Option<Box<dyn SettlementClient>>,
+    /// The database client
+    database: Option<Box<dyn Database>>,
+    /// Queue client
+    queue: Option<Box<dyn QueueProvider>>,
+    /// Storage client
+    storage: Option<Box<dyn DataStorage>>,
+}
+
+#[cfg(test)]
+impl Default for TestConfigBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[cfg(test)]
+impl TestConfigBuilder {
+    /// Create a new config
+    pub fn new() -> TestConfigBuilder {
+        TestConfigBuilder {
+            starknet_client: None,
+            da_client: None,
+            prover_client: None,
+            settlement_client: None,
+            database: None,
+            queue: None,
+            storage: None,
+        }
+    }
+
+    pub fn mock_da_client(mut self, da_client: Box<dyn DaClient>) -> TestConfigBuilder {
+        self.da_client = Some(da_client);
+        self
+    }
+
+    pub async fn build(mut self) -> MockServer {
+        dotenvy::from_filename("../.env.test").expect("Failed to load the .env file");
+
+        // init starknet client
+        if self.starknet_client.is_none() {
+            let provider = JsonRpcClient::new(HttpTransport::new(
+                Url::parse(get_env_var_or_panic("MADARA_RPC_URL").as_str()).expect("Failed to parse URL"),
+            ));
+            self.starknet_client = Some(Arc::new(provider));
+        }
+
+        // init database
+        if self.database.is_none() {
+            self.database = Some(Box::new(MongoDb::new(MongoDbConfig::new_from_env()).await));
+        }
+
+        // init queue
+        if self.queue.is_none() {
+            self.queue = Some(Box::new(SqsQueue {}));
+        }
+
+        // init the DA client
+        if self.da_client.is_none() {
+            self.da_client = Some(build_da_client().await);
+        }
+
+        let settings_provider = DefaultSettingsProvider {};
+
+        // init the Settings client
+        if self.settlement_client.is_none() {
+            self.settlement_client = Some(build_settlement_client(&settings_provider).await);
+        }
+
+        // init the Prover client
+        if self.prover_client.is_none() {
+            self.prover_client = Some(build_prover_service(&settings_provider));
+        }
+
+        // init the storage client
+        if self.storage.is_none() {
+            self.storage = Some(build_storage_client().await);
+        }
+
+        // return config and server as tuple
+        let config = Config::new(
+            self.starknet_client.unwrap(),
+            self.da_client.unwrap(),
+            self.prover_client.unwrap(),
+            self.settlement_client.unwrap(),
+            self.database.unwrap(),
+            self.queue.unwrap(),
+            self.storage.unwrap(),
```

Comment:
To avoid these unwraps, a better pattern is to do, for each entry (example with `prover_client`):

```
let prover_client = self.prover_client.unwrap_or_else(|| build_prover_service(&settings_provider));
```

and then just pass `prover_client` to `Config::new()`.

Reply:
Valid, `unwrap_or_else` can readily be used for `prover_client` for better code pattern.

Upon further inspection : 
- settlement_client
- storage_client
- da_client
- database
are built using `async build functions` which makes it tricky to create using `unwrap_or_else`.

We can write something like : 

```rust
use tokio::runtime::Handle;

// ...

self.settlement_client = Some(self.settlement_client.unwrap_or_else(|| {
    Handle::current().block_on(build_settlement_client(&settings_provider))
}));

```
But then we'll have unnecessary runtime overhead.

### Proposed Solution : 
 - change `sync build functions` to use `unwrap_or_else`.
 - handle `async build functions` in separate assignment using `await`.
 
 WDYT ?